Whiskey/Functions/New-WhiskeyContext.ps1

```diff
@@ -119,16 +119,14 @@ function New-WhiskeyContext
 
     $packageJsonPath = Join-Path -Path $buildRoot -ChildPath 'package.json'
     $ignorePackageJsonVersion = $config.ContainsKey('IgnorePackageJsonVersion') -and $config['IgnorePackageJsonVersion']
-    if( -not $config.ContainsKey('Version') -and (Test-Path -Path $packageJsonPath -PathType Leaf) -and -not $ignorePackageJsonVersion )
+    if( $config.ContainsKey( 'VersionPath') )
     {
-        $config['Version'] = Get-Content -Raw -Path $packageJsonPath | ConvertFrom-Json | Select-Object -ExpandProperty 'version' -ErrorAction Ignore
-        if( $config['Version'] -eq '0.0.0' )
-        {
-            [void]$config.Remove('Version')
-        }
+        $semVersion = New-WhiskeySemanticVersion -Path $config['VersionPath'] -Prerelease $prereleaseInfo -BuildMetadata $buildMetadata -ErrorAction Stop

```

Comment:
Don't call `New-WhiskeySemanticVersion` twice. Instead, setup a hashtable that contains the parameters you want to pass:

```
$versionParam = @{}
if( $config.ContainsKey('VersionFrom') )
{
    $versionParam['Path'] = $config['VersionFrom']
}
else
{
    $versionParam['Version'] = $config['Version']
}

$semVersion = New-WhiskeySemanticVersion @versionParam -Prerelease $prereleaseInfo -BuildMetadata $buildMetadata -ErrorAction Stop

Reply:
Done